src/parser/Statement.ts

```diff
@@ -95,6 +99,20 @@ export class Body extends Statement {
         return result;
     }
 
+    getTypedef(state: TranspileState) {
```

Comment:
Why not having a "transpilation mode", even as a static flag somewhere, to control that instead of doubling half the transpilation logic with those `getTypedef` methods?

Reply:
I considered that, but most expressions and statements will not emit any typedef information. Having a separate function allows me to detect which items we should even walk in the first place (and avoid needless `transpile` recursing).  See 
`if ('getTypedef' in statement) {` a few lines below this comment.

Also, if most of the transpile function can be reused, I'm already calling it within the `getTypedef` method so we don't actually duplicate any logic.  (see [comment statement](https://github.com/rokucommunity/brighterscript/blob/4be26ae9e5691446d4bb81338e4490260338ff44/src/parser/Statement.ts#L278-L280)). Meanwhile, something like a  [NamespaceStatement](https://github.com/rokucommunity/brighterscript/blob/4be26ae9e5691446d4bb81338e4490260338ff44/src/parser/Statement.ts#L1093-L1110) doesn't actually transpile itself, so that's where having a separate function keeps the logic clean between the two concepts (`transpile` and `getTypedef`). Or ClassStatement, whose `transpile` function is extremely complex and doesn't have _any_ similarities with what its `typedef` would look like.